Apps/Common/SIMSolverAdap.h

```diff
@@ -16,6 +16,39 @@
 
 #include "SIMSolver.h"
 #include "AdaptiveSIM.h"
+#include "TimeStep.h"
+
+
+/*!
+  \brief Adaptive simulator driver using the ISolver interface.
+*/
+
+template<class T1>
+class AdaptiveISolver : public AdaptiveSIM
+{
+public:
+  //! \brief The constructor initializes default adaptation parameters.
+  //! \param sim The FE model
+  //! \param sa If \e true, this is a stand-alone driver
+  AdaptiveISolver(T1& sim, bool sa = true) : AdaptiveSIM(sim, sa), model(sim) {}
+
+protected:
+  //! \brief Assemble and solve the equation system.
+  virtual bool assembleAndSolveSystem()
+  {
+    TimeStep dummy;
+    model.init(dummy);
+    bool result = model.solveStep(dummy);
+    if (result)
+      solution = model.getSolutions();
+
+    rCond = 1.0;
```

Comment:
I don't quite see why you have to do this assignment. Isn't it better to set  a sufficient loose tolerance on the condition number if you don't want to terminate the adaptive loop because of that? Then also `rCond` can remain private to AdaptiveSIM, as the other control parameters. Or what did I miss here.